staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -470,12 +478,13 @@ func (c *AvailableConditionController) getAPIServicesFor(obj runtime.Object) []*
 			ret = append(ret, apiService)
 		}
 	}
+	if svc != nil {
+		c.cache[svc.ObjectMeta.GenerateName] = ret
```

Comment:
this cache grows without bound, and stores a zero-length entry for every service, which we do not want

Reply:
I was thinking about this as well.
In deleteService, after the call to getAPIServicesFor, we can remove the entry for the corresponding key.
What do you think ?

Reply:
we only want items in the cache that actually map to apiservices. everything else should use the zero value of the map as an indication there are no apiservices for that namespace/name

Reply:
I think the casting above:
```
	if svc, ok = obj.(*v1.Service); ok {
```
would satisfy that the item maps to apiservice